components/camel-opentracing/src/main/java/org/apache/camel/opentracing/decorators/RestSpanDecorator.java

```
@@ -0,0 +1,108 @@
+import org.apache.camel.Exchange;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestSpanDecorator extends AbstractHttpSpanDecorator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RestSpanDecorator.class);
+
+    @Override
+    public String getComponent() {
+        return "rest";
+    }
+
+    @Override
+    public String getOperationName(Exchange exchange, Endpoint endpoint) {
+        return getPath(endpoint.getEndpointUri());
+    }
+
+    @Override
+    public void pre(Span span, Exchange exchange, Endpoint endpoint) {
+        super.pre(span, exchange, endpoint);
+
+        getParameters(getPath(endpoint.getEndpointUri())).forEach(param -> {
+            Object value = exchange.getIn().getHeader(param);
+            if (value != null) {
+                if (value instanceof String) {
+                    span.setTag(param, (String)value);
+                } else if (value instanceof Number) {
+                    span.setTag(param, (Number)value);
+                } else if (value instanceof Boolean) {
+                    span.setTag(param, (Boolean)value);
+                }
+            }
+        });
+        
+    }
+
+    protected static String getPath(String uri) {
+        // Obtain the 'path' part of the URI format: rest://method:path[:uriTemplate]?[options]
+        String path = null;
+        int index = uri.indexOf(':');
+        if (index != -1) {
+            index = uri.indexOf(':', index + 1);
+            if (index != -1) {
+                path = uri.substring(index + 1);
+                index = path.indexOf('?');
+                if (index != -1) {
+                    path = path.substring(0, index);
+                }
+                path = path.replaceAll(":", "");
+                try {
+                    path = URLDecoder.decode(path, "UTF-8");
+                } catch (UnsupportedEncodingException e) {
+                    LOG.error("Failed to decode URL path '" + path + "'", e);
```

Dont log at ERROR as there is nothing the user can do. Log a DEBUG or something and say in the error message that this exception is ignored.